### DISCUSSION TITLE
How to resolve "Request entity too large" error ?

### DISCUSSION THREAD

**Discussion by BilluMeow** [Unleash/unleash#6958] [Q&A] [answered]
I am trying to add many tenants (users) in strategy constraint using this [API](https://docs.getunleash.io/reference/api/unleash/update-feature-strategy) . The request become too large and it returns the error "Request entity too large".

I cannot update strategy in multiple requests as if I don't provide the previous tenants, unleash will remove them from the strategy.

What alternate options do I have ?  I am using unleash v5 (open source).

**Comment by FredrikOseberg** [comment, marked as answer]
Hi @BilluMeow 

Apologies for the late response. The restrictions are in place because Unleash is not intended to be a data storage. Adding many users into strategies is considered an anti-pattern because it increases the size of the payloads transmitted and feature toggle evaluations in your application. We have written the following document to suggest alternative approaches to achieving the same thing: https://docs.getunleash.io/understanding-unleash/managing-constraints

If this doesn't help, please continue the discussion here.
